src/arch/host/include/arch/atomic.h

```diff
@@ -32,6 +32,27 @@ static inline void arch_atomic_init(atomic_t *a, int32_t value)
 	arch_atomic_set(a, value);
 }
 
+#if defined __XCC__
+static inline int32_t arch_atomic_add(atomic_t *a, int32_t value)
+{
+	int32_t result, current;
+
+	current = arch_atomic_read(a);
+	result = current + value;
+	arch_atomic_set(a, result);
+	return current;
+}
+
+static inline int32_t arch_atomic_sub(atomic_t *a, int32_t value)
```

Comment:
How is this different from `arch_atomic_add(a, -value);`?

Slightly different undefined overflow? :-)

Reply:
Yes, it's a shame the C standard didn't define signed overflows as 2's complement. It generates unnecessary code just for possible existence of theoretical different integer ALU implementations.